From 67d02c0d4b5435f5d13eca8a18ca14ac4630225f Mon Sep 17 00:00:00 2001
From: Subrata Chatterjee <schatterjee@habana.ai>
Date: Thu, 28 Jul 2022 10:17:41 +0300
Subject: [PATCH] [SW-98285] gaudi2: ras: add mss_cls unit reset

VLSI suggestion is to do MSS_CLS_UNIT_RST along with
current MME_EUS_UNIT_RST after MBIST to avoid ECC.

This patch adds MSS_CLS_UNIT_RST also in the unit reset
after mbist function.

Signed-off-by: Subrata Chatterjee <schatterjee@habana.ai>
Change-Id: I8ef614f24e734bd988207ee534ff4b489eb08159
---

diff --git a/arch/gaudi2/arc/asic.c b/arch/gaudi2/arc/asic.c
index c57b3b0..29f89a1 100644
--- a/arch/gaudi2/arc/asic.c
+++ b/arch/gaudi2/arc/asic.c
@@ -1122,6 +1122,7 @@
 	regmap_writel(PSOC_RESET_CONF_SM_UNIT_RST_EN_MASK, mmPSOC_RESET_CONF_SM_UNIT_RST);
 	regmap_writel(PSOC_RESET_CONF_HBM_MC_UNIT_RST_EN_MASK, mmPSOC_RESET_CONF_HBM_MC_UNIT_RST);
 	regmap_writel(PSOC_RESET_CONF_MME_EUS_UNIT_RST_EN_MASK, mmPSOC_RESET_CONF_MME_EUS_UNIT_RST);
+	regmap_writel(PSOC_RESET_CONF_MSS_CLS_UNIT_RST_EN_MASK, mmPSOC_RESET_CONF_MSS_CLS_UNIT_RST);
 	regmap_writel(PSOC_RESET_CONF_NIC_UNIT_RST_EN_MASK, mmPSOC_RESET_CONF_NIC_UNIT_RST);
 	regmap_writel(PSOC_RESET_CONF_NIC_PRT_UNIT_RST_EN_MASK, mmPSOC_RESET_CONF_NIC_PRT_UNIT_RST);
 	regmap_writel(PSOC_RESET_CONF_SRAM_UNIT_RST_EN_MASK, mmPSOC_RESET_CONF_SRAM_UNIT_RST);
@@ -1134,6 +1135,7 @@
 	regmap_writel(0, mmPSOC_RESET_CONF_ROTATOR_UNIT_RST);
 	regmap_writel(0, mmPSOC_RESET_CONF_HBM_MC_UNIT_RST);
 	regmap_writel(0, mmPSOC_RESET_CONF_MME_EUS_UNIT_RST);
+	regmap_writel(0, mmPSOC_RESET_CONF_MSS_CLS_UNIT_RST);
 	regmap_writel(0, mmPSOC_RESET_CONF_NIC_UNIT_RST);
 	regmap_writel(0, mmPSOC_RESET_CONF_NIC_PRT_UNIT_RST);
 	regmap_writel(0, mmPSOC_RESET_CONF_SRAM_UNIT_RST);
